### PR TITLE
Fix first zap when modal closed (#771)

### PR DESCRIPTION
**Pull request by jp30566347**
## Description

Fixes the onShortPress callback not being invoked after the long press callback is.

 - ~Assign a new key to force remounting of LongPressable component when modal is closed from long press~
 - Update LongPressable component with custom version with a fix (See https://github.com/stardust66/react-longpressable/pull/14) 
 - Extract handlers
 - Remove unnecessary async keyword from callback
 - Remove hover state when closing modal, otherwise it stays colored

## Screenshots

https://github.com/jp30566347/stacker.news/assets/1215807/e96be5ce-eab6-4bbc-9ba3-f48f4faf11dd

## Additional Context

~Could not find the root cause of this, but this workaround seems to trigger proper callback registration.~

I tried:
 - created a sandboxed environment from the react-longpressable and could NOT reproduce
 - removed children entirely (`<ActionTooltip />`) and this was still happening~

Turns out @huumn had good instinct on the modal messing with LongPressable event handlers.

## Checklist

- [x] Are your changes backwards compatible?

- [x] Did you QA this? Could we deploy this straight to production?

- [x] For frontend changes: Tested on mobile?

- [ ] Did you introduce any new environment variables? If so, call them out explicitly in the PR description.

### CODE REVIEW

File: components/upvote.js [hunk 1]
```diff
@@ -142,43 +143,51 @@ export default function UpVote ({ item, className }) {
...
+
+  const handleLongPress = (e) => {
+    if (!item) return
+
+    // we can't tip ourselves
+    if (disabled) {
+      return
+    }
+
+    // force remount to fix onClick handler being ignored
```
Comment (ekzyis) [inline components/upvote.js:158]: +1 for the comment

**Review by ekzyis — APPROVED** [review]
Can confirm this fixes the issue. Don't see anything wrong with this approach except that we don't fully understand where it's coming from.

However, imo, bug is definitely annoying enough to justify fixing it this way instead of digging deeper. +1 for the additional context provided.

Additional context from me:

- Interesting that this only happens on long press. If I don't have enough sats, the QR code for payment pops up on short press. If I close it, short press still works as expected on first try.
- It's only broken for the item that was clicked. If you long press and then close, short press works for any other item on first try except the item was long pressed before.

**Comment by huumn** [comment]
I remember debugging this when I first noticed it after it was introduced (years ago). 

IIRC it's caused by the modal popping up when the long press threshold is reached. The modal prevents `onPointerUp` from running which sets `isLongPressing = false`. So when the modal closes `isLongPressing = true` still.

**Comment by jp30566347** [comment]
Good hint!

I wonder what is the reason behind this line: https://github.com/stardust66/react-longpressable/blob/99e33d25e7c720a91eb259eb1b4b72a2bec8837d/src/index.js#L76

I think it should be turned off now that the callback is getting invoked for long press?

**Comment by huumn** [comment]
> I think it should be turned off now that the callback is getting invoked for long press?

Definitely! I think it's just to control whether the shortpress or the longpress handler is called `onPointerUp`.

**Comment by jp30566347** [comment]
I'll confirm and get back to you

**Comment by jp30566347** [comment]
Opened a PR to their repo: https://github.com/stardust66/react-longpressable/pull/14

In the meantime, happy to fork under stacker.news org and use that version.

It seems to fix it for me 🤞

**Comment by huumn** [comment]
If you want to just move your version to components and uninstall the package, that'd be great! (It's a pretty tiny package.)

**Pull request merged**